bfs_battle_for_supremacy/game_logic/managers/map_manager.py

```diff
@@ -1,9 +1,135 @@
+                else:
+                    print("Invalid selection. Please select a valid object.")
+            else:
+                print("Invalid selection. Square is empty.")
+
+        elif MapManager.selection_counter == 1:
+            if MapManager.check_availability(square):
+                MapManager.target_position = square
+                MapManager.selection_counter = 2
+                print(
+                    "Selected target position:"
+                    + f"({square.row}, {square.column})"
+                )
+                MapManager.move_with_bfs(current_player)
+            else:
+                print(
+                    "Target position is not empty."
+                    + "Please select an empty square."
+                )
+
+        else:
+            print("Too many selections. Resetting selection.")
+            MapManager.reset_selection()
+
+    @staticmethod
+    def bfs_path_finding(start_square, target_square):
+        rows = len(MapManager.game_map.grid)
+        cols = len(MapManager.game_map.grid[0])
+
+        directions = [(-1, 0), (1, 0), (0, -1), (0, 1)]
+        visited = set()
+        queue = deque([(start_square, [])])
+
+        while queue:
+            current_square, path = queue.popleft()
+
+            if current_square == target_square:
+                return path + [target_square]
+
+            if current_square in visited:
+                continue
+
+            visited.add(current_square)
+
+            for direction in directions:
+                next_row = current_square.row + direction[0]
+                next_col = current_square.column + direction[1]
+
+                if 0 <= next_row < rows and 0 <= next_col < cols:
+                    next_square = MapManager.game_map.get_square(
+                        next_row, next_col
+                    )
+                    if (
+                        MapManager.check_availability(next_square)
+                        and next_square not in visited
+                    ):
+                        queue.append((next_square, path + [current_square]))
+
+        return []
+
+    @staticmethod
+    def move_with_bfs(current_player):
+        if not MapManager.current_position or not MapManager.target_position:
+            print("Positions not set. Cannot move object.")
+            return
+
+        path = MapManager.bfs_path_finding(
+            MapManager.current_position, MapManager.target_position
+        )
+
+        if not path:
+            print("No valid path found.")
+            MapManager.reset_selection()
+            return
+
+        distance = len(path) - 1
+
+        if distance > MapManager.movement_counter:
+            print(
+                "Move exceeds remaining movement points "
+                + f"({MapManager.movement_counter})."
+            )
+            MapManager.reset_selection()
+            return
+
+        item = MapManager.current_object
+        for step in path[1:]:
```

Comment:
this should be delayed in the background (if the steps were done in a for like this the whole app will wait for it and will display the final movement only).

use asyncio.call_later() or a Timer for scheduling each move and run the whole thing in background.